src/components/basic/display/MarketPrice/viewer.tsx

```diff
@@ -0,0 +1,66 @@
+import React, { useMemo, memo } from "react";
+import styled from "styled-components";
+
+import { Text, Icon, Loader } from "@gnosis.pm/safe-react-components";
+
+import { TokenDetails } from "types";
+
+import { SubtextAmount } from "components/basic/display/SubtextAmount";
+import { TokenDisplay } from "components/basic/display/TokenDisplay";
+import { Tooltip } from "components/basic/display/Tooltip";
+import { Link } from "components/basic/inputs/Link";
+
+const Amount = styled.span`
+  display: grid;
+  grid-template-columns: repeat(5, auto);
+  column-gap: 0.25em;
+  align-items: center;
+`;
+
+export interface MarketPriceViewerProps {
+  price?: string;
+  isPriceLoading: boolean;
+  priceUrl: string;
+  baseTokenAddress?: string;
+  quoteTokenAddress?: string;
+}
+
+const _MarketPriceViewer = (props: MarketPriceViewerProps): JSX.Element => {
+  const {
+    price,
+    isPriceLoading,
+    priceUrl,
+    baseTokenAddress,
+    quoteTokenAddress,
+  } = props;
+
+  const amount = useMemo(
+    (): JSX.Element | string =>
+      isPriceLoading ? (
```

Comment:
Double nested shorthand ifs caused problems for me in the past. Can you move this to a var with normal ifs?